apps/src/dance/lab2/ProgramExecutor.ts

```diff
@@ -115,7 +116,13 @@ export default class ProgramExecutor {
     }
 
     this.hooks.runUserSetup();
-    this.nativeAPI.p5_.draw();
+
+    // force preview draw to occur **after** any
+    // draw iterations already queued up
+    setTimeout(() => {
```

Comment:
I could do this in a promise if preferable, although I find setTimeout more readable for simple stuff like this. For anyone interested, I ended up doing a little research ([SO here](https://stackoverflow.com/questions/38752620/promise-vs-settimeout)) on the distinction between using setTimeout and a promise that immediately resolves.

Reply:
I think that as long as the comment is included, either is fine.

Reply:
Interesting that there is a last draw call is from the call to`noLoop` from `nativeAPI.reset` - thanks for sharing that this doc! https://p5js.org/reference/#/p5/draw.